ports/raspberrypi/common-hal/wifi/Radio.c

```diff
@@ -55,6 +56,15 @@
 #define PING_ID        0xAFAF
 #endif
 
+#ifndef PING_DEBUG
+#define PING_DEBUG     LWIP_DBG_ON
+#endif
+
+#ifdef LWIP_DEBUG
+static u32_t ping_time;
```

Comment:
```suggestion
static uint32_t ping_time;
```
Thanks! @bill88t, just one minor suggestion.

Reply:
Note: It is copied from `ports/raspberrypi/lwip_src/ping.c`, line 98-100.

Reply:
In C, "static" means that the scope of the thing (`ping_time`) is restricted to the single source file. So, when debug is enabled there's one "ping_time_ variable in Radio.c, and it's separate from the one in "lwip_src/ping.c". The print in Radio.c will always be based on a ping_time of 0, so it'll show incorrect results.

Reply:
(that's one meaning of static, it has others :rofl: but that's the meaning when it's used to declare a variable at function scope)

Reply:
I didn't check for how `ping_time` was being used in the code, now that I have taken a look at the implementation, I think it would be better to just move the custom `pin_recv` function to `ports/raspberrypi/lwip_src/ping.c`.

Reply:
Umm `ports/raspberrypi/lwip_src/ping.c` is unused.
I renamed it and the build was successful.

Reply:
Removed source files are a class of problem not detected by the build system, that is, if "foo.o" has been created and "foo.c" is subsequently removed, the build system does not signal a problem, it just re-uses "foo.o".

According to this test, the file is used
```
jjepler@bert:~/src/circuitpython/ports/raspberrypi$ echo '#error check if unused' > lwip_src/ping.c 
jepler@bert:~/src/circuitpython/ports/raspberrypi$ make
I: Building for raspberry_pi_pico_w 0 
Use make V=1, make V=2 or set BUILD_VERBOSE similarly in your environment to increase build verbosity.
QSTR not updated
lwip_src/ping.c:1:2: error: #error check if unused
    1 | #error check if unused
      |  ^~~~~
make: *** [../../py/mkrules.mk:62: build-raspberry_pi_pico_w/lwip_src/ping.o] Error 1
```

Reply:
I always clean before making. I don't care if I lose time like this.

I run am running `make V=0 BOARD=raspberry_pi_pico_w clean && make -j4 V=1 DEBUG=1 BOARD=raspberry_pi_pico_w`.
(I know I could just set the variables once)

Commit incoming.

Reply:
> Umm `ports/raspberrypi/lwip_src/ping.c` is unused.

Hmm... does it use `ports/raspberrypi/lib/lwip` submodule?
If so we should probably delete the `lwip_src` folder in order to avoid confusion.

Reply:
yes, lib/lwip is used

lwip_src exists specifically so that files from lwip that unavoidably had to be modified didn't require us to change the submodule. It's nice if we can reduce this to 0.

Git does not track directories per se, so if after an update there are no tracked files in the directory, and no untracked files in the directory, I think git removes the empty directory. Likewise in a fresh clone it will not create the directory in the first place.

Reply:
The `lwip_src` stuff should stay for now..
Some other stuff is using `ping.c` and without `ping.h` some stuff are acting sus on actions
(but not on local somehow even though I `make clean`).
Not willing to pursue removing them in this pr. I may try it at another time.
For now, this is considered solved, as `ping_time` is residing within `ping.h` and works as expected.